sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -310,8 +315,84 @@ public <T> Mono<CosmosItemResponse<T>> createItem(T item, CosmosItemRequestOptio
         return withContext(context -> createItemInternal(item, requestOptions, context));
     }
 
+    private <T> Mono<CosmosItemResponse<T>> createItemWithRetriesInternal(
+        T item, CosmosItemRequestOptions options) {
+
+        return createItemInternal(item, options)
+            .onErrorResume(throwable -> {
+                Throwable error = throwable instanceof CompletionException ? throwable.getCause() : throwable;
+
+                if (!(error instanceof CosmosException)) {
+
+                    Exception nonCosmosException =
+                        error instanceof Exception ? (Exception) error : new RuntimeException(error);
+                    return Mono.error(nonCosmosException);
+                }
+
+                assert error instanceof CosmosException;
+                CosmosException cosmosException = (CosmosException) error;
+
+                if (cosmosException.getStatusCode() != HttpConstants.StatusCodes.CONFLICT) {
+                    return Mono.error(cosmosException);
+                }
+
+                InternalObjectNode internalObjectNode = InternalObjectNode.fromObjectToInternalObjectNode(item);
+                String itemId = internalObjectNode.getId();
+                RequestOptions requestOptions = ModelBridgeInternal.toRequestOptions(options);
+                @SuppressWarnings("unchecked")
+                Class<T> itemType = (Class<T>) item.getClass();
+
+                Mono<CosmosItemResponse<T>> readMono =
+                    this.getDatabase().getDocClientWrapper()
+                        .readDocument(getItemLink(itemId), requestOptions)
+                        .map(response -> {
+                            CosmosDiagnostics responseDiagnostics = response.getDiagnostics();
+                            if (responseDiagnostics != null &&
+                                responseDiagnostics.getClientSideRequestStatisticsRaw() != null) {
+
+                                CosmosDiagnostics errorDiagnostics = cosmosException.getDiagnostics();
+                                if (errorDiagnostics != null) {
+                                    responseDiagnostics.clientSideRequestStatistics().recordContributingPointOperation(
```

Comment:
Generally the response statics list is recorded by timestamp, but seems the merge behavior behind this will put the latest response diagnostics first?

Reply:
No specific ordering - I can add that

Reply:
Done